modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/java/com/liferay/content/dashboard/document/library/internal/upgrade/registry/v_1_17/AssetVocabularyUpgradeProcess.java

```diff
@@ -0,0 +1,108 @@
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.content.dashboard.document.library.internal.upgrade.registry.v_1_17;
+
+import com.liferay.asset.kernel.model.AssetVocabulary;
+import com.liferay.asset.kernel.service.AssetVocabularyLocalService;
+import com.liferay.portal.kernel.repository.model.FileEntry;
+import com.liferay.portal.kernel.service.ClassNameLocalService;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.ArrayUtil;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portlet.asset.util.AssetVocabularySettingsHelper;
+
+import java.util.Arrays;
+
+/**
+ * @author Cristina González
+ */
+public class AssetVocabularyUpgradeProcess extends UpgradeProcess {
+
+	public AssetVocabularyUpgradeProcess(
+		AssetVocabularyLocalService assetVocabularyLocalService,
+		ClassNameLocalService classNameLocalService,
+		CompanyLocalService companyLocalService) {
+
+		_assetVocabularyLocalService = assetVocabularyLocalService;
+		_classNameLocalService = classNameLocalService;
+		_companyLocalService = companyLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		_companyLocalService.forEachCompany(
+			company -> {
+				AssetVocabulary audienceAssetVocabulary =
+					_assetVocabularyLocalService.fetchGroupVocabulary(
+						company.getGroupId(),
+						StringUtil.toLowerCase(
+							GetterUtil.getString("audience")));
+
+				if (audienceAssetVocabulary != null) {
+					_updateAssetVocabulary(audienceAssetVocabulary);
+				}
+
+				AssetVocabulary stageAssetVocabulary =
+					_assetVocabularyLocalService.fetchGroupVocabulary(
+						company.getGroupId(),
+						StringUtil.toLowerCase(GetterUtil.getString("stage")));
+
+				if (stageAssetVocabulary != null) {
+					_updateAssetVocabulary(stageAssetVocabulary);
+				}
+			});
+	}
+
+	private AssetVocabulary _updateAssetVocabulary(
+			AssetVocabulary assetVocabulary)
+		throws Exception {
+
+		long classNameId = _classNameLocalService.getClassNameId(
+			FileEntry.class.getName());
+
+		AssetVocabularySettingsHelper assetVocabularySettingsHelper =
+			new AssetVocabularySettingsHelper(assetVocabulary.getSettings());
+
+		long[] classNameIds = assetVocabularySettingsHelper.getClassNameIds();
+
+		if (ArrayUtil.contains(classNameIds, classNameId)) {
+			long[] selectedClassNameIds = ArrayUtil.remove(
```

Comment:
@cgoncas, instead of removing, wouldn't it be better to replace for DLFileEntry?

Reply:
Hey @jkappler !!

The upgrade process do it after the upgrade process has been executed :).

So we don't need to do it here explicitly :).

Thanks!!